src/raftstore/store/fsm/peer.rs

```diff
@@ -713,15 +749,55 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
         if self.fsm.peer.is_applying_snapshot() || self.fsm.peer.has_pending_snapshot() {
             // need to check if snapshot is applied.
             self.fsm.has_ready = true;
+            self.fsm.missing_ticks = 0;
             self.register_raft_base_tick();
             return;
         }
+
+        let mut res = None;
+        if self.ctx.cfg.hibernate_regions {
+            if self.fsm.group_state == GroupState::Idle {
+                if self.fsm.missing_ticks + 1 < self.ctx.cfg.raft_election_timeout_ticks {
+                    self.register_raft_base_tick();
+                    self.fsm.missing_ticks += 1;
+                }
+                return;
+            }
+            res = Some(self.fsm.peer.check_before_tick(&self.ctx.cfg));
+            if self.fsm.missing_ticks > 0 {
+                for _ in 0..self.fsm.missing_ticks {
+                    if self.fsm.peer.raft_group.tick() {
+                        self.fsm.has_ready = true;
+                    }
+                }
+                self.fsm.missing_ticks = 0;
+            }
+        }
         if self.fsm.peer.raft_group.tick() {
```

Comment:
Is it safe to tick here?

Reply:
Why is it unsafe? That's the very reason why tick is registered.